src/materials/nodes/manager/NodeMaterialObserver.js

```diff
@@ -362,6 +363,13 @@ class NodeMaterialObserver {
 		const storedAttributeNames = Object.keys( storedAttributes );
 		const currentAttributeNames = Object.keys( attributes );
 
+		if ( storedGeometryData.id !== geometry.id ) {
```

Comment:
To clarify this does not introduce a "ping-pong" effect since the `storedGeometryData` is maintained per render object.